src/server-manager/grpc-gateway.ts

```diff
@@ -217,29 +221,54 @@ export class GrpcGateway extends EventEmitter {
     request.setRequestid(requestId)
 
     try {
+      if (this.timeoutNumber > 10 && Date.now() - this.startTime <= 3 * 60 * 1000) {
```

Comment:
We shouldn't put the check at the beginning of the request, we should do a reconnect when we found the last timeout happened.

Reply:
Done!